### ISSUE TITLE
Support for "source" of OBGenericData

### ISSUE THREAD

**Issue by openbabel-bot** [openbabel/openbabel#1467]
Posted from Noel's e-mail:

As I understand it, the idea behind the DataTypes is exactly this, to
distinguish different types of data from each other. Could we have a
particular data type for data fields contained in a molecular file
format, and make it separate from data types that are computed by
OpenBabel?

For example, for an SD file with two data fields, Molecular weight and
NSC, I find the following OBGenericDatas:
('Comment', 'CORINA 2.61 0041  25.10.2001', 3)
('NSC', '1', 1)
('Molecular weight', '122.03677944', 1)]

(i.e. Attribute, Value, Datatype)

However, if I compute the SSSR and access GetFormula, in addition I get:
('RingData', 'RingData', 8)
('Formula', 'C7H6O2', 1)

Note that although RingData has its own datatype, 8, Formula is thrown
in with the data fields from the SD file.

In summary, if it makes sense to everyone, could computed datatypes be
kept separate from those that are parsed from the file. Otherwise, it
will be impossible to distinguish data fields from calculated fields.

Reported by: @ghutchis

Original Ticket: "openbabel/feature-requests/98":https://sourceforge.net/p/openbabel/feature-requests/98
